elastic/kibana
# [Ingest Pipelines] Fix description field bug

## Summary

Fixed a bug where the description of a processor would vanish after editing the processor in the processor form flyout.

Also added slight rendering optimisation for the text field that was needlessly re-rendering on scroll because of how selectors where being calculated on every scroll.

## How to test

1. Go to Ingest Pipelines app
2. Create a new pipeline (or edit one)
3. Give a processor a description (after creating one or editing an existing one)
4. Edit the processor settings, for example the "Field" field
5. Update the processor and see the description vanish

After the fix included on line 84 of `x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/processor_form/processor_form.container.tsx` the description should not vanish.

The problem stems from the fact that the form does not provide a way to edit description, but the description lives on the form options object which gets re-calculated after submitting the form and does not include the processor description. The fix for now is to re-add the processor description on submit.

## Gif

![ingest-nodes-proc-description-disappear](https://user-images.githubusercontent.com/8155004/95079313-b9a1dd80-0716-11eb-934c-2d762098750c.gif)
